### PR TITLE
Get rid of panic in stats/ fix dir lxc

### PR DESCRIPTION
**Pull request by jessfraz**

### CODE REVIEW

**Comment by jfrazelle** [comment]
ping @crosbymichael

**Comment by jfrazelle** [comment]
ugh wait testing more

**Comment by jfrazelle** [comment]
ok ya we good

File: daemon/daemon.go [hunk 1]
```diff
@@ -913,7 +913,8 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine) (*Daemon, error)
 		selinuxSetDisabled()
 	}
 
-	daemonRepo := path.Join(config.Root, "containers")
+	const runDir = "/var/run/docker"
+	daemonRepo := path.Join(runDir, "containers")
```
Comment (LK4D4) [inline daemon/daemon.go]: But that's not quite right I think. That data should survive restart.
ping @crosbymichael

File: daemon/execdriver/lxc/driver.go [hunk 2]
```diff
@@ -783,5 +783,8 @@ func (d *driver) Exec(c *execdriver.Command, processConfig *execdriver.ProcessCo
 }
 
 func (d *driver) Stats(id string) (*execdriver.ResourceStats, error) {
+	if _, ok := d.activeContainers[id]; !ok {
+		return &execdriver.ResourceStats{}, fmt.Errorf("%s is not a key in active containers", id)
```
Comment (LK4D4) [inline daemon/execdriver/lxc/driver.go]: better return nil instead.

File: daemon/daemon.go (hunk 1 above)
Comment (jfrazelle) [inline daemon/daemon.go]: problem was on lxc it was frekaing out about not being able to find /var/run/docker/containers, so maybe its referencing that somewhere else will look

Comment (jfrazelle) [inline daemon/daemon.go]: ah hang on i got it

Comment (jfrazelle) [inline daemon/daemon.go]: its line 702,

**Comment by jfrazelle** [comment]
updated its not pretty

**Comment by jfrazelle** [comment]
boom got it, gitblame lead to https://github.com/docker/docker/commit/8dc5791f7302aea0734a26dc1f54d96b04771b8f which makes sense because the panic was **NOT** fixed in lxc omg my life i literally cant even ping @LK4D4 @crosbymichael

**Comment by tiborvass** [comment]
LGTM

**Comment by jfrazelle** [comment]
Ah man 1 stupid test on lxc failed I'll fix in a bit

**Comment by jfrazelle** [comment]
it's unrelated tho

**Comment by duglin** [comment]
LGTM

**Pull request merged**